src/Canvas.cc

```diff
@@ -768,6 +770,14 @@ NAN_METHOD(Canvas::RegisterFont) {
   g_free(style);
 }
 
+NAN_METHOD(Canvas::DeregisterAllFonts) {
+	// Unload all fonts from pango to free up memory
+	std::for_each(font_face_list.begin(), font_face_list.end(), [&](FontFace& f) {
+      deregister_font( (unsigned char *)f.file_path );
+    });
+	font_face_list.clear();
```

Comment:
I think this is probably a memory leak. We need to free up the PangoFontDescriptions held by the FontFaces. It would be great if we did that in the `FontFace` destructor.

Reply:
Yup, you're right.  So, I added a `FontFace` destructor as you requested, and freed `user_desc` and `sys_desc` if they were non-null, but this resulted in a segmentation fault.  I tried to track it down but couldn't find the cause.  I ended up just freeing the memory in the `DeregisterAllFonts` function, which works fine on both my test platforms.  I hope this is acceptable.